utils/variant_qc.py

```diff
@@ -0,0 +1,346 @@
+    :param BooleanExpression proband_is_female_expr: An optional expression giving the sex the proband. If not given, DNMs are only computed for autosomes.
+    :return: An expression with the counts
+    :rtype: StructExpression
+    """
+
+    # Create map for transmitted, untransmitted and DNM
+    hom_ref = 0
+    het = 1
+    hom_var = 2
+
+    auto_or_par = 2
+    hemi_x = 1
+    hemi_y = 0
+
+    trans_config_counts = {
+        # kid, dad, mom, copy -> t, u
+        (hom_ref, het, het, auto_or_par): (0, 2),
+        (hom_ref, hom_ref, het, auto_or_par): (0, 1),
+        (hom_ref, het, hom_ref, auto_or_par): (0, 1),
+        (het, het, het, auto_or_par): (1, 1),
+        (het, hom_ref, het, auto_or_par): (1, 0),
+        (het, het, hom_ref, auto_or_par): (1, 0),
+        (het, hom_var, het, auto_or_par): (0, 1),
+        (het, het, hom_var, auto_or_par): (0, 1),
+        (hom_var, het, het, auto_or_par): (2, 0),
+        (hom_var, het, hom_var, auto_or_par): (1, 0),
+        (hom_var, hom_var, het, auto_or_par): (1, 0),
+        (hom_ref, hom_ref, het, hemi_x): (0, 1),
+        (hom_ref, hom_var, het, hemi_x): (0, 1),
+        (hom_var, hom_ref, het, hemi_x): (1, 0),
+        (hom_var, hom_var, het, hemi_x): (1, 0)
+    }
+
+    trans_count_map = hl.literal(trans_config_counts)
+
+    def _get_copy_state(locus: hl.expr.LocusExpression) -> hl.expr.Int32Expression:
+        """
+        Helper method to go from LocusExpression to a copy-state int for indexing into the
+        trans_count_map.
+        """
+        return (
+            hl.case()
+            .when(locus.in_autosome_or_par(), auto_or_par)
+            .when(locus.in_x_nonpar(), hemi_x)
+            .when(locus.in_y_nonpar(), hemi_y)
+            .or_missing()
+        )
+
+    def _get_composite_filter_expr(
+            expr1: hl.expr.BooleanExpression,
+            expr2: Optional[hl.expr.BooleanExpression]
+    ) -> hl.expr.BooleanExpression:
+        """
+        Helper method to join two expression with support for None.
+        """
+        if expr2 is None:
+            return expr1
+        else:
+            return expr1 & expr2
+
+    def _is_dnm(
+            proband_gt: hl.expr.CallExpression,
+            father_gt: hl.expr.CallExpression,
+            mother_gt: hl.expr.CallExpression,
+            locus: hl.expr.LocusExpression,
+            proband_is_female: Optional[hl.expr.BooleanExpression]
+    ) -> hl.expr.BooleanExpression:
+        """
+        Helper method to get whether a given genotype combination is a DNM at a given locus with a given proband sex.
+        """
+        if proband_is_female is None:
+            logger.warning("Since no proband sex expression was given to generate_fam_stats_expr, only DNMs in autosomes will be counted.")
+            return hl.or_missing(
+                locus.in_autosome(),
+                trio_mt.proband_entry.GT.is_het() & trio_mt.father_entry.GT.is_hom_ref() & trio_mt.mother_entry.GT.is_hom_ref()
```

Comment:
Can this be `proband_gt.is_het() & father_gt.is_hom_ref() & mother_gt.is_hom_ref()` like below, or is there a reason that it needs to use the trio_mt?

Reply:
it _should_ be `proband_gt.is_het() & father_gt.is_hom_ref() & mother_gt.is_hom_ref()` ! Thanks for catching this -- typical refactoring fatigue :)